specs/phase0/p2p-interface.md

```diff
@@ -856,6 +856,10 @@ Clients MUST support requesting blocks since the latest finalized epoch.
 Clients MUST respond with at least one block, if they have it.
 Clients MAY limit the number of blocks in the response.
 
+Clients MAY consider including a block in the response as soon as it it passes the gossip validation rules.
+Clients MUST NOT respond with blocks that failed gossip validation rules.
+Clients MUST NOT respond with blocks that failed `fork_choice.on_block`.
```

Comment:
I have mixed feeling on this.
I remember @djrtwo mentioning to avoid serving blocks that turns out to be invalid, but I might recall incorrectly.

An argument in favour is that it seems odd to have clients serving blocks that turned out to be invalid for them.
As the time pass by, the client can from better ideas on a particular block, so makes sense to me that most recent byRoot responses leverage these additional evaluations applied after initial gossip rules.

Reply:
yeah, I don't think it's a MUST NOT. I think it's a MAY NOT ( you can safely prune these) but I'm not 100% sure if it needs to be called out.

Also, I'm a fan of the state transition being hte filter instead of the on_block as on_block has many timing and stateful conditions here

Reply:
> Also, I'm a fan of the state transition being hte filter instead of the on_block as on_block has many timing and stateful conditions here

I see your point. I only see a potential issue due to the fact that in `on_block` as written in spec, `is_data_available`  comes before the state transition, and if we want clients keep serving blocks with unavailable blobs, seems like they could do that even without checking the state transition. But clients implementations are most likely not following that order, so let's change..

>  I don't think it's a MUST NOT. I think it's a MAY NOT
agreed.